sdk/monitor/Microsoft.OpenTelemetry.Exporter.AzureMonitor/src/TelemetryPartA.cs

```diff
@@ -0,0 +1,92 @@
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using OpenTelemetry.Resources;
+using OpenTelemetry.Trace;
+using Microsoft.OpenTelemetry.Exporter.AzureMonitor.Models;
+
+namespace Microsoft.OpenTelemetry.Exporter.AzureMonitor
+{
+    internal class TelemetryPartA
+    {
+        private static readonly IReadOnlyDictionary<TelemetryType, string> PartA_Name_Mapping = new Dictionary<TelemetryType, string>
+        {
+            [TelemetryType.Request] = "Request",
+            [TelemetryType.Dependency] = "RemoteDependency",
+            [TelemetryType.Message] = "Message",
+            [TelemetryType.Event] = "Event",
+        };
+
+        internal static string RoleName { get; set; } = null;
+
+        internal static string RoleInstance { get; set; } = null;
+
+        internal static TelemetryItem GetTelemetryItem(Activity activity, string instrumentationKey)
+        {
+            TelemetryItem telemetryItem = new TelemetryItem(PartA_Name_Mapping[activity.GetTelemetryType()], activity.StartTimeUtc.ToString(CultureInfo.InvariantCulture))
+            {
+                InstrumentationKey = instrumentationKey
+            };
+
+            InitRoleInfo(activity);
+            telemetryItem.Tags[ContextTagKeys.AiCloudRole.ToString()] = RoleName;
+            telemetryItem.Tags[ContextTagKeys.AiCloudRoleInstance.ToString()] = RoleInstance;
+            telemetryItem.Tags[ContextTagKeys.AiOperationId.ToString()] = activity.TraceId.ToHexString();
+
+            if (activity.Parent != null)
+            {
+                telemetryItem.Tags[ContextTagKeys.AiOperationParentId.ToString()] = activity.Parent.SpanId.ToHexString();
+            }
+
+            telemetryItem.Tags[ContextTagKeys.AiInternalSdkVersion.ToString()] = SdkVersionUtils.SdkVersion;
+
+            return telemetryItem;
+        }
+
+        internal static void InitRoleInfo(Activity activity)
+        {
+            if (RoleName != null || RoleInstance != null)
+            {
+                return;
+            }
+
+            var resource = activity.GetResource();
+
+            if (resource == null)
+            {
+                return;
+            }
+
+            string serviceName = null;
+            string serviceNamespace = null;
+
+            foreach (var attribute in resource.Attributes)
+            {
+                if (attribute.Key == Resource.ServiceNameKey && attribute.Value is string)
+                {
+                    serviceName = attribute.Value.ToString();
+                }
+                else if (attribute.Key == Resource.ServiceNamespaceKey && attribute.Value is string)
+                {
+                    serviceNamespace = attribute.Value.ToString();
+                }
+                else if (attribute.Key == Resource.ServiceInstanceIdKey && attribute.Value is string)
+                {
+                    RoleInstance = attribute.Value.ToString();
+                }
```

Comment:
Do you know which is more common; a) an attribute Key other than these expected three values or b)  attribute value other than a string?

I think the duplicate `is string` check could be simplified to an outer if. but maybe that's not necessary if we expect several different possible keys. 

Also, it stands out that after we identify that the type is a string we call ToString().
I think this could be simplified as `if (attribute.Value is string attributeValue)`.

Reply:
In OpenTelemetry latest version has `Resource.ServiceNameKey`, `Resources.*` constants were removed. This complete functionality of this method will change when we move to RC version of OpenTelemetry. I will leave the code in this way and will rework when refining `InitRoleInfo` method sooner.

Reply:
Yes. Resource will no longer be part of Activity, and instead will be accessible from Exporter directly.